server/models/ClientRequest.js

```diff
@@ -183,26 +184,32 @@ class ClientRequest {
     });
   }
 
-  checkHash(options) {
-    const hashes = this.getEnsuredArray(options.hashes);
-    const stoppedHashes = hashes.filter(hash => {
-      return torrentService.getTorrent(hash).status.includes(torrentStatusMap.stopped);
-    });
-    const hashesToStart = [];
+  checkHash(userId, options) {
+    const torrentService = ServicesHandler.getTorrentService(userId);
+    torrentService.fetchTorrentList().then(
+      () => {
+        const hashes = this.getEnsuredArray(options.hashes);
+        const stoppedHashes = hashes.filter(hash => {
```

Comment:
There's no need to create this array; we can just check `torrentService.getTorrent(hash)` in the `forEach`.

Reply:
`torrentService` could have been loaded but haven't yet fetched the torrent list (since it doesn't call `fetchTorrentList()` in the constructor)